firmware/defmt-itm/src/lib.rs

```diff
@@ -0,0 +1,76 @@
+//! [`defmt`](https://github.com/knurling-rs/defmt) global logger over RTT.
+//!
+//! NOTE when using this crate it's not possible to use (link to) the `rtt-target` crate
+//!
+//! To use this crate, link to it by importing it somewhere in your project.
+//!
+//! ```
+//! // src/main.rs or src/bin/my-app.rs
+//! use defmt_rtt as _;
+//! ```
+
+#![no_std]
+
+use core::{
+    ptr::NonNull,
+    sync::atomic::{AtomicBool, Ordering},
+};
+
+use cortex_m::{interrupt, itm, peripheral::ITM, register};
+
+static ENABLED: AtomicBool = AtomicBool::new(false);
+
+/// Enables defmt logging over the ITM stimulus port 0
```

Comment:
```suggestion
/// Enables defmt logging over the ITM stimulus port 0.
///
/// This needs to be called by the application before defmt logging is used, otherwise the logs will be disposed.
```

Could we do this automatically, perhaps? Or would that have any problematic side-effects?

Reply:
you mean writing to `ITM.TER` lazily? that can be done
what I wanted to about though is the user sending unrelated data over the stimulus port 0 (using e.g. `iprintln!`) as that would corrupt the defmt data stream

Reply:
Ah, right. Seems fine then.